### PR TITLE
add coverage to `UserCommons.java`

### LINKED ISSUE #20: 6pm-4:  Add tests for UserCommons Controller
- [x] `UserCommons.java` has 100% test coverage
- [x] `UserCommons.java` has full mutation coverage

### PR DESCRIPTION
**Pull request by rsdharmaji**
# Overview

this PR adds full test coverage to `UserCommons.java`. 

# Issues Addressed

closes #20 

# Details

test coverage was added.

### CODE REVIEW

**Review by sytarno — DISMISSED** [review]
Looks good. I fixed your comparison logic, since it led to some failing tests.

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java
Comment (btk5h) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java]: Can we remove the commented code here?

File: src/main/java/edu/ucsb/cs156/happiercows/entities/UserCommons.java
Comment (btk5h) [inline src/main/java/edu/ucsb/cs156/happiercows/entities/UserCommons.java]: Should this live in the main codebase? I think it makes more sense to move this somewhere under the test directory to avoid cluttering the main file and to prevent the method from accidentally being used in production code.

**Review by btk5h — CHANGES_REQUESTED** [review]

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java
Comment (btk5h) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java]: Why is this logic here? Is the ordering of the JSON properties inconsistent? It might be better to use the [`responseToJson`](https://github.com/ucsb-cs156-w22/team04-w22-6pm-HappyCows/blob/main/src/test/java/edu/ucsb/cs156/happiercows/ControllerTestCase.java#L34) method, as this workaround will not scale for larger JSON response bodies.

**Review by btk5h — DISMISSED** [review]

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java
Comment (rsdharmaji) [reply src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java]: done

File: src/main/java/edu/ucsb/cs156/happiercows/entities/UserCommons.java
Comment (rsdharmaji) [reply src/main/java/edu/ucsb/cs156/happiercows/entities/UserCommons.java]: refactored into a method in the test file

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java
Comment (rsdharmaji) [reply src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java]: `responseToJson` is used now instead

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java
Comment (jacksoncooper) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java:89]: I saw that Evan wrote the assertions against the `expectedJson` object. For example,

```java
assertEquals(expectedJson.message, "UserCommons with commonsId 1 and userId 1 not found")
```

This will crash the test if `expectedJson` doesn't have the desired members, but I think it's more readable.

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java
Comment (jacksoncooper) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java:126]: Same thing here. It's up to your preference.

**Review by jacksoncooper — COMMENTED** [review]
Just stylistic stuff.

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java
Comment (rsdharmaji) [reply src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java:126]: i think it's fine as is.

Comment (jacksoncooper) [reply src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java:126]: Okay.

**Review by jacksoncooper — APPROVED** [review]
No complaints.

**Review by sytarno — APPROVED** [review]
Basically the same as mine. I like the way you used responseToJson. Coverage seems to be 100%, so no issues there either. Good job implementing changes at the request of Bryan.

**Review by pconrad — APPROVED** [review]
LGTM.. just waiting on CI/CD again after merging in latest main branch changes

**Review by pconrad — APPROVED** [review]
LGTM

**Comment by pconrad** [comment]
Mergeable.. just waiting on backend CI/CD

**Pull request merged**
